scripts/reindex.py

```diff
@@ -0,0 +1,114 @@
+# purpose of this script is to rebuild a new solr collection
+# it will automatically activate it (by swapping cores)
+
+import os 
+import sys
+import pickle
+import requests
+import time
+from subprocess import PIPE, Popen
+
+
+homedir = os.path.dirname(os.path.dirname(__file__))
+lockfile = os.path.abspath(homedir + '/reindex.lock')
+solr_url = 'http://localhost:9983/solr'
+admin_url = '%s/admin/cores' % solr_url
+update_url = '%s/collection2/update' % solr_url
+
+def run():
```

Comment:
this code will mostly run via a cronjob.  does it make sense to use both print statements and logger to get the info to kibana?

Reply:
I have incorporated your comments from above. As for the last point, I think it is a good idea to run the rebuild when maxDocs reaches numDocs + X (the X being number of deleted docs; currently with 4M we were getting into issues - so perhaps X=2M?)

You should decide what is the most appropriate course of action.